### PR TITLE
feat: add function to fetch locked balances for given lockId

### LINKED ISSUE #57: Add fetching of locked balances by lockId and address
We need to add fetching of locked balances due to how the LBP `repayTarget` works, and how vesting is implemented.

Knowing what the locked balance of a given address is, is detrimental to figuring out e.g. which LBP fee should be applied.

### Detailed Description
Locked balances exist in two palettes, both `balances` and `tokens`. Where `balances` will include locks for the native asset, and `tokens` will include locks for 3rd party assets. Each lock is identified by a `lockId` which can be used to determine the purpose/reason of the balance lock.

### Possible Implementation

Example query:
```graphql
query GetLockedBalanceById($address: String!, $lockId: String) {
    lockedBalance(address: $address, lockId: $lockId) {
        id,
        amount
    }
}
```

Native asset locks can be fetched as follows:
`balances.locks(accountId)`

Balances for non-native assets can be fetched as follows:
`tokens.accounts(accountId)`

#### Handling of `lockId`

Since the node storage does not offer filtering by ID, we have to filter for the correct `lockId` on the client side, by filtering out the unwated lock entries for both native and non native assets. The resolver should only return the lock which matches the requested `lockId`. If locks for both assets are present and match the given `lockId`, the native asset lock takes precedence.

> Ignoring one of the locks in a case there are multiple locks for the given `lockId` is not an ideal solution, but to date we don't have a feature which would create a lock for both assets using the same `lockId`. This can change in the future.

### PR DESCRIPTION
**Pull request by dexterslabor**
Closes #57

### CODE REVIEW

**Comment by mckrava** [comment]
Storybook testing artifacts:
- [storybook-ci-tests-results](https://github.com/galacticcouncil/Basilisk-ui/suites/5328745389/artifacts/166147920);
Or check GH Action `Storybook testing CI flow` for more logs.

**Comment by mckrava** [comment]
:page_with_curl: **Basilisk UI reporter** <br /> _Report has been triggered by commit [Merge branch 'feat/lbp-v1' into feat/locked-balances (8e2841bb07e63b5f919b8ceee5a02df06e86b7bf)](https://github.com/galacticcouncil/Basilisk-ui/commit/8e2841bb07e63b5f919b8ceee5a02df06e86b7bf)_ <hr />:small_blue_diamond: **App/Storybook build:** <br />    - Status: :no_entry_sign: _Failed_ <hr />:small_blue_diamond: **App/Storybook deployment:** <br />    - Status: :no_entry_sign: _Failed_

**Comment by mckrava** [comment]
This should be fixed - https://github.com/galacticcouncil/Basilisk-ui/runs/5733831757?check_suite_focus=true#step:8:18

File: src/hooks/balances/resolvers/query/lockedBalances.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    if (!args.lockId) throw Error(errors.missingArgumentsLockedBalanceQuery);
+    // can't fetch lockedBalance without address
+    if (!args.address && !obj.id)
+      throw Error(errors.missingArgumentsLockedBalanceQuery);
+
+    const address = args.address ? args.address : obj.id;
+
+    return (
+      await getLockedBalancesByLockId(apiInstance, address, args.lockId)
+    ).map((lockedBalance: LockedBalance) => {
```
Comment (maht0rz) [inline src/hooks/balances/resolvers/query/lockedBalances.tsx:48]: Can we turn this into a `withId`, same as we use `withTypename`?

**Review by maht0rz — APPROVED** [review]

**Pull request merged**